mvp_example/src/main/java/com/distillery/android/blueprints/mvp/adapter/ToDoListAdapter.kt

```diff
@@ -0,0 +1,57 @@
+package com.distillery.android.blueprints.mvp.adapter
+
+import android.graphics.Paint.STRIKE_THRU_TEXT_FLAG
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.distillery.android.ui.databinding.ItemTodoBinding
+import com.distillery.android.domain.models.ToDoModel
+
+class ToDoListAdapter(
+    private val onDeleteClickListener: (toDoModel: ToDoModel) -> Unit,
+    private val onCompleteClickListener: (toDoModel: ToDoModel) -> Boolean
+) :
+        ListAdapter<ToDoModel, ToDoListAdapter.ToDoViewHolder>(ToDoListDiffCallback()) {
+
+    class ToDoViewHolder(private val binding: ItemTodoBinding) : RecyclerView.ViewHolder(binding.root) {
+        fun bind(
+            toDoModel: ToDoModel,
+            onDeleteClickListener: (toDoModel: ToDoModel) -> Unit,
+            onCompleteClickListener: (toDoModel: ToDoModel) -> Boolean
+        ) {
```

Comment:
The class can be `inner` to avoid passing listeners to individual holders:
```suggestion
    inner class ToDoViewHolder(private val binding: ItemTodoBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(toDoModel: ToDoModel) {
```